------------------------------------------------------------
revno: 14068
revision-id: squid3@treenet.co.nz-20160723071930-cemledcltg8pkc28
parent: squid3@treenet.co.nz-20160723071620-1wzqpbyi1rk5w6vg
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4542
author: Anonymous <bigparrot@pirateperfection.com>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Sat 2016-07-23 19:19:30 +1200
message:
  Bug #4542: authentication credentials IP TTL updated incorrectly
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20160723071930-cemledcltg8pkc28
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: ee0c6aab5414532d9554ef338cce049263902fd8
# timestamp: 2016-07-23 07:24:05 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20160723071620-\
#   1wzqpbyi1rk5w6vg
# 
# Begin patch
=== modified file 'src/auth/User.cc'
--- src/auth/User.cc	2016-01-01 00:14:27 +0000
+++ src/auth/User.cc	2016-07-23 07:19:30 +0000
@@ -284,7 +284,7 @@
             /* This ip has already been seen. */
             found = 1;
             /* update IP ttl */
-            ipdata->ip_expiretime = squid_curtime;
+            ipdata->ip_expiretime = squid_curtime + ::Config.authenticateIpTTL;
         } else if (ipdata->ip_expiretime <= squid_curtime) {
             /* This IP has expired - remove from the seen list */
             dlinkDelete(&ipdata->node, &ip_list);

